### PR TITLE
APPS-2578 Nextflow Pipeline Applet script - refactor for readability

### PR DESCRIPTION
**Pull request by r-i-v-a**
Details, testing on Jira - APPS-2578

### CODE REVIEW

**Comment by Gvaihir** [comment]
Review WIP, but regarding the tests - where are those? Or are you only referring to the regressions?

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 1]
```diff
@@ -15,48 +18,183 @@ MAX_WAIT_AFTER_JOB_ERROR=240
...
 
-  jq '.docker_registry.token' "$CREDENTIALS" -r | docker login $REGISTRY --username $REGISTRY_USERNAME --password-stdin 2> >(grep -v -E "WARNING! Your password will be stored unencrypted in |Configure a credential helper to remove this warning. See|https://docs.docker.com/engine/reference/commandline/login/#credentials-store")
-  if [ ! $? -eq 0 ]; then
-      echo "Docker authentication failed, please check if the docker credentials file is correct." 1>&2
-      exit 2
+  # set default NXF env constants
+
+  # Disable use of newer flag --cpus when running Docker
+  # Can be enabled when Docker version on DNAnexus workers supports it
+  export NXF_DOCKER_LEGACY=true
```
[the remark this answers is not in the document]
Comment (mhrvol) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:53]: https://docs.docker.com/engine/release-notes/prior-releases/#1131-2017-02-08

It should be part of our worker docker already, according to @wormsik we have 23.0.6 (last year May/June).

I created  APPS-2668 to test this.

**Comment by r-i-v-a** [comment]
> Review WIP, but regarding the tests - where are those? Or are you only referring to the regressions?

Just links to runs of existing tests (this isn't supposed to change behavior, only reorganize).

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 2]
```diff
@@ -1,7 +1,10 @@
 #!/usr/bin/env bash
 
-# Script that is used to run both the main Nextflow orchestrator job
-# and the Nextflow task sub-jobs.
+# =========================================================
+# Script with entrypoints to run
+# 1. Nextflow head (orchestrator) job
+# 2. Nextflow task subjobs
```
[the remark this answers is not in the document]
Comment (r-i-v-a) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh]: done

**Review by mhrvol — APPROVED** [review]
lgtm

**Review by Gvaihir — APPROVED** [review]
Approving provided the tests are passing

**Review by YuxinShi0423 — APPROVED** [review]
Looks great!

File: src/python/dxpy/templating/templates/nextflow/src/nextflow.sh [hunk 3]
```diff
@@ -284,6 +443,30 @@ setup_workdir() {
...
+        echo "Profile was given in run options... overriding the default profile ($profile_arg)"
+        profile_arg=""
+      fi
+      ;;
+    *) ;;
+    esac
+  done
+}
+
 dx_path() {
```
[the remark this answers is not in the document]
Comment (r-i-v-a) [reply src/python/dxpy/templating/templates/nextflow/src/nextflow.sh:470]: Thanks for highlighting additional cleanup items, I'll make tasks for those @YuxinShi0423

**Pull request merged**